keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1065,6 +1066,61 @@ Details to consider:
 
 #### CEL Expression Composition
 
+##### Match Conditions
+
+Note that the syntax of the `matchConditions` resource is intended to
+align with the [Admission Webhook Match Conditions KEP #3716](https://github.com/kubernetes/enhancements/pull/3717),
+so that KEP should be controlling with regard to deviations in the schema.
+This section is focused specifically on how the `matchConditions` concept can
+be applied to in-process admission.
+
+The match criteria in bindings are not expected to be able to cover all possible
+ways users may want to scope their policies. For example, there is no way to
+match off of kind, only resource. To provide extensibility for the match criteria
+without requiring modifying every validation rule individually, a global predicate
+system is needed. These predicates contain CEL statements that must be satisfied, otherwise
+the policy will be ignored. In order to keep bindings language-agnostic and to support
+singleton policies, the logic should live in the policy definition resource. To enable
+customization per-binding, the CEL statements should have access to the parameter resource.
+
+Here is an example of a policy definition using policy predicates (under the `matchConditions` field):
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "replicalimit-policy.example.com"
+Spec:
+  failurePolicy: Fail
+  paramKind:
+    apiVersion: rules.example.com/v1
+    kind: ReplicaLimit
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+  matchConditions:
+    - expression: 'metadata.kind == "Deployment"'
```

Comment:
https://github.com/kubernetes/enhancements/pull/3717/files#r1100587323

> I suggest you make these {name, expression} pairs. That way it would be easy to have a controller own and maintain one or more named expressions. The name helps explain what the expression is for, and it make SSA work well. The name could also be reported/logged in errors etc.

Reply:
I like the idea. I'll follow whatever gets decided on the linked KEP

Reply:
Added